index.html

```diff
@@ -3325,6 +3323,48 @@ <h2>
       <h2>
         Security Considerations
       </h2>
+      <section class="informative">
+        <h2 id="sec-enhancements">
+          Mechanisms to enhance security
+        </h2>
+        <p>
+          This specification includes a number of mechanisms to enhance the
+          security of the API:
+        </p>
+        <ul data-link-for="PaymentRequest">
+          <li>Explicit user action is required to call <a>PaymentRquest</a>'s
+          <a>show()</a> method. This reduces content's ability to unexpectedly
+          cause a request for payment to be displayed to the end-user.
+          </li>
+          <li>Interfaces are <a data-cite=
+          "WEBIDL#dfn-available-only-in-secure-contexts">available only in a
+          secure context</a>, to reduce the possibly that credentials will be
+          acquired and transferrer through insecure communication protocols.
+          </li>
+          <li>A <a>top-level browsing context</a> need to explicitly grant an
+          <a>iframe</a> the ability to access the <a>PaymentRequest</a>
+          interface via the <a>allowpaymentrequest</a> attribute. This prevents
+          embedded third-party content from accessing the interfaces of the
+          <cite>Payment Request API</cite> without the <a>top-level browsing
+          context</a>'s permission.
+          </li>
+          <li>In the definition of <a>canMakePayment()</a> the Working Group
+          seeks a balance between user experience and date protection. As
+          defined, <a>canMakePayment()</a> provides the party that calls the
+          API with information about the user's environment. To reduce the
+          potential for abuse, implementers plan a number of mitigations,
+          including rate-limiting <a>canMakePayment()</a> calls from the same
+          origin.
+          </li>
+          <li>A user agent can limit matching (in <a>show()</a> and
```

Comment:
As above... I'm not entirely sure we adequately solved this or are mitigating attacks adequately yet.

Reply:
As indicated in @lknik's review, rate-limiting sounds nice but any legitimate or illegitimate actor can easily jump over that hurdle.

Reply:
It is my understanding that implementations are doing this. @adrianba, @zkoch, @rsolomakhin do you have any experience to share or any new thoughts on this topic?